### ISSUE TITLE
`cml-send-comment` comments are removed upon rebase

### ISSUE THREAD

**Issue by SaulAryehKohn** [iterative/cml#735]
When a branch with a PR is rebased, the commits are orphaned and the comments from `cml-send-comment` are deleted. Is there a way to preserve them after a rebase? Alternatively, is there a way for `cml-send-comment` to comment on the PR itself, rather than a certain commit?

**Comment by 0x2b3bfa0** [comment]
> _Is there a way to preserve them after a rebase?_

None that is known to me, at least.

> _Alternatively, is there a way for `cml-send-comment` to comment on the PR itself, rather than a certain commit?_

We still don't support this feature, and rely on commit comments to display the same information at the commit and the pull request levels.

**Comment by 0x2b3bfa0** [comment]
@SaulAryehKohn & @saadlaaribi, feel free to request pull request comments (as opposed to commit comments) as a feature if you deem it useful.

**Comment by DavidGOrtega** [comment]
👋 @SaulAryehKohn ```cml-send-comment``` has an option ```update``` 

> Update the last CML comment (if any) instead of creating a new one

This feature will update always your last cml comment that ideally could be the very first commit in the branch. Not sure if it fits all your needs but maybe is worth a try

**Comment by SaulAryehKohn** [comment]
>  @SaulAryehKohn `cml-send-comment` has an option `update`
> 
> > Update the last CML comment (if any) instead of creating a new one
> 
> This feature will update always your last cml comment that ideally could be the very first commit in the branch. Not sure if it fits all your needs but maybe is worth a try

:eyes: I will try that, thank you. I think all comments linked to commits are going to be wiped once the rebase alters the commit hash itself though. Stable solution will be to post the comment to the PR rather than to the commit. @0x2b3bfa0 could you please let me know how to formally request it as a feature?

**Comment by DavidGOrtega** [comment]
@SaulAryehKohn this feature might not be feasible. Not because its not possible in GH but maybe is not possible in other SCM. Just to peek a bit more. The idea would be using the body of a PR instead?

**Comment by SaulAryehKohn** [comment]
@DavidGOrtega yes, a good reference is this action: https://github.com/EnricoMi/publish-unit-test-result-action which stays at the PR level but contains per-commit results

**Comment by 0x2b3bfa0** [comment]
> _The idea would be using the body of a PR instead?_

Either that, or just like `cml send-comment` but with pull request comments instead of commit comments. If I recall correctly it's supported on every provider: `cml send-comment --commit-sha=···` would use commit comments (that propagate to the pull request), and `cml send-comment --pr=···` would use pull request comments. 🤔 

> _@0x2b3bfa0 could you please let me know how to formally request it as a feature?_

If you believe it would be an useful feature, please open a separate issue proposing it.

**Comment by DavidGOrtega** [comment]
Aha! PRs are sharing the issues API. I guess its doable.
```--pr=123``` should be inferred. Probably getting the PRs of a branch.
